### ISSUE TITLE
Add tomcat system property "catalina.base"?

### ISSUE THREAD

**Issue by chanlewis** [ran-jit/tomcat-cluster-redis-session-manager#24]
How can i do if i want to deploy multiple tomcats on one server with different ports ? 
Such as , /home/tomcat-8080  /home/tomcat-8081  ...
Can i use the same file (redis-data-cache.properties) ?

**Comment by ran-jit** [comment]
Yep, you've to use the same redis config details in all the tomcat nodes.

**example:**
* /home/tomcat-8080/conf/redis-data-cache.properties
* /home/tomcat-8081/conf/redis-data-cache.properties

Thanks!

**Comment by ran-jit** [comment]
Sorry @chanlewis . Please ignore my previous comment.

The system read the redis-data-cache config properties based on the "catalina.base" system property.

example: **catalina.base="/home/tomcat-8080"**, all the tomcat servers read the redis-data-cache config properties from **/home/tomcat-8080/conf** directory.

Thanks!

**Comment by chanlewis** [comment]
Dear Ranjith Manickam,


All right, I was just going to talk about this issue.
Set the "catalina.base" system property in /etc/profile or ~/.bash_profile once.
All the tomcat nodes read the same redis-data-cache config properties.


Thank you very much!
Chan Lewis


At 2018-12-28 20:05:26, "Ranjith Manickam" <notifications@github.com> wrote:


Sorry @chanlewis . Please ignore my previous comment.

The system read the redis-data-cache config properties based on the "catalina.base" system property.

example: catalina.base="/home/tomcat-8080", all the tomcat servers read the redis-data-cache config properties from /home/tomcat-8080/conf directory.

Thanks!

**Comment by ran-jit** [comment]
Thanks Chan Lewis.

Closing this issue. Please feel-free to reopen this issue/create new issue, if you require additional details in future.

**Comment by chanlewis** [comment]
Dear Ranjith Manickam,

That's all right!
Excuse me, I have to reopen this issue, because I find that dot character  does not supported on linux server system property .
For example "export catalina.base=/opt/tomcat" does not supported on CentOS, Debian, ubantu .
So, I suggest you to change the variable to "CATALINA_BASE" in SessionConstants.java . Thank you!




chan_lewis@163.com
 
From: Ranjith Manickam
Date: 2018-12-28 21:59
To: ran-jit/tomcat-cluster-redis-session-manager
CC: 刘程; Mention
Subject: Re: [ran-jit/tomcat-cluster-redis-session-manager] Add tomcat system property "catalina.base"? (#24)
Thanks Chan Lewis.
Closing this issue. Please feel-free to reopen this issue/create new issue, if you require additional details in future.

**Comment by chanlewis** [comment]
Dear Ranjith Manickam,

I'm so sorry! That i had a mistake understanding about system property. 
Please ignore my previous mail.




chan_lewis@163.com
 
From: Ranjith Manickam
Date: 2018-12-28 21:59
To: ran-jit/tomcat-cluster-redis-session-manager
CC: 刘程; Mention
Subject: Re: [ran-jit/tomcat-cluster-redis-session-manager] Add tomcat system property "catalina.base"? (#24)
Thanks Chan Lewis.
Closing this issue. Please feel-free to reopen this issue/create new issue, if you require additional details in future.
